packages/grafana-data/src/utils/processDataFrame.ts

```diff
@@ -44,6 +44,23 @@ function convertTimeSeriesToDataFrame(timeSeries: TimeSeries): DataFrame {
   };
 }
 
+function convertGraphSeriesToDataFrame(graphSeries: GraphSeriesXY): DataFrame {
```

Comment:
Not really sure about this one.

Reply:
Good point, seems like we even revert it back to legacydata:
https://github.com/grafana/grafana/blob/8b0bbb2c4f316fe5140f585100fb5f268a5532f6/public/app/plugins/datasource/prometheus/components/PromQueryField.tsx#L178

I'll see if I can get rid of this.

Reply:
I've looked into this but I can't find an easy fix for it. I think we've to introduce PanelData for the Explore state and that seems to be a bigger PR so I'll keep this for now.
The problem is this line https://github.com/grafana/grafana/blob/b3a75cee680d90dd51cd8bc41ebade51a10636f5/public/app/features/explore/QueryRow.tsx#L220
that needs to be solved.

Reply:
yes, think we need to add the raw DataFrame result to the ExploreState as going from GraphXYSeries back to DataFrame can't be good :)

Reply:
Created this issue to be worked on later: #18528